src/frontend/components/SelectContent/index.tsx

```diff
@@ -0,0 +1,430 @@
+              fit="cover"
+              fill="vertical"
+              src={content?.urls.thumbnails['480']}
+            />
+          ) : (
+            <IconBox className="icon-file-text2" />
+          )}
+        </CardBody>
+        <CardHeader
+          pad={{ horizontal: 'small', vertical: 'small' }}
+          background="#000000A0"
+          width="large"
+          justify="start"
+        >
+          <Box>
+            <Heading level="3">{content?.title}</Heading>
+            <Box gap="small" direction="row" align="end">
+              <IconStatus
+                assertion={content?.upload_state === 'ready'}
+                trueMessage={intl.formatMessage(
+                  messages.uploaded,
+                  contentTitle,
+                )}
+                falseMessage={intl.formatMessage(
+                  messages.notUploaded,
+                  contentTitle,
+                )}
+                TrueIcon={DocumentUpload}
+                FalseIcon={DocumentMissing}
+              />
+
+              <IconStatus
+                assertion={content?.is_ready_to_show}
+                trueMessage={intl.formatMessage(
+                  messages.readyToShow,
+                  contentTitle,
+                )}
+                falseMessage={intl.formatMessage(
+                  messages.notReadyToShow,
+                  contentTitle,
+                )}
+                TrueIcon={Monitor}
+                FalseIcon={Monitor}
+              />
+            </Box>
+          </Box>
+        </CardHeader>
+      </Stack>
+    </Card>
+  );
+};
+
+const SelectContentSection = ({
+  title,
+  addMessage,
+  newTitle,
+  newLtiUrl,
+  emptyMessage,
+  items,
+  selectUrl,
+}: {
+  title: string;
+  addMessage: MessageDescriptor;
+  newTitle: MessageDescriptor;
+  newLtiUrl: string;
+  emptyMessage: MessageDescriptor;
+  items: Nullable<[Video | Document]>;
+  selectUrl: any;
+}) => {
+  const intl = useIntl();
+  const size = React.useContext(ResponsiveContext);
+
+  return (
+    <React.Fragment>
+      <Box direction="row">
+        <Heading level="2">{title}</Heading>
+        <Button
+          color={'brand'}
+          label={<FormattedMessage {...addMessage} />}
+          onClick={() => selectUrl(newLtiUrl, intl.formatMessage(newTitle))}
+        />
+      </Box>
+
+      {items && items.length > 0 ? (
+        <Grid
+          columns={size !== 'small' ? 'medium' : '100%'}
+          gap="medium"
+          rows="medium"
+        >
+          {items?.map((item, index) => (
+            <ContentCard
+              content={item}
+              key={index}
+              onClick={() => selectUrl(item!.lti_url, item!.title)}
+            />
+          ))}
+        </Grid>
+      ) : (
+        <Box>
+          <FormattedMessage {...emptyMessage} />
+          <Anchor label={<FormattedMessage {...addMessage} />} />
+        </Box>
+      )}
+    </React.Fragment>
+  );
+};
+
+interface SelectContentProps {
```

Comment:
In this interface all properties can not exists. But when you use them in the component, you assert they exist, this interface is wrong IMO and does not reflect how you use it.